config/application.rb

```diff
@@ -72,20 +72,265 @@ class Application < Rails::Application
-      c = Settings.action_mailer
-      (c[:default_url_options] ||= {})[:host] ||= Settings.hostname
-      c.each do |key,val|
-        config.action_mailer.send("#{key}=", val)
+    # Include the external mappings
+    f = File.join(Rails.root, 'config', 'external_link_mapping.yml')
+    config.external_url_mapping = APP_CONFIG = YAML.load(File.read(f))
+
+
+    # Check Settings and replace invalid/add required values using
+    # reasonable defaults when needed.
+    tmp = Settings.hostname ? Settings.hostname.to_s : ''
+    i = tmp.index(':')
+    if i
+      port = tmp[i..-1]			# yes, we take the ':'
+      tmp = tmp[0..i-1]
+    else
+      port = ''
+    end
+    begin
+      if tmp.length < 1
+        if Rails.env == 'production'
+          tmp = Socket.gethostname
+          port = Rails.env != 'production' ? ':3000' : ''
+        else
+          tmp = 'localhost'
+          port = ':3000'
+        end
+      end
+    rescue
+      tmp = nil
+    end
+    if tmp && tmp != 'localhost' && ! tmp.index('.')	# need fqdn?
+      begin
+        tmp = Addrinfo.tcp(tmp, 0).getnameinfo.first
+      rescue
+        tmp = nil
+      end
+    end
+    if ! tmp
+      raise 'Unable to determine the FQDN of this machine. Please add an ' \
+        'appropriate "hostname: your_fqdn" to the config/settings.local.yml ' \
+        'file and try again.'
+    end
+    Settings.hostname = tmp + port
+    config.fqdn = tmp
+    config.port = port[1..-1]
+
+    if Settings.log_level
+      tmp = Settings.log_level.to_s
+      if tmp.in? ['fatal', 'error', 'warn', 'info', 'debug']
+        config.log_level = tmp.intern
+      else
+        STDERR.puts "Invalid setting log_level '#{tmp}' ignored"
+      end
+    end
+
+    tmp = Settings.consider_all_requests_local
+    config.consider_all_requests_local = tmp && tmp.to_s == 'true'
+
+    # If one has configured it, make sure that at least all recipients are ok
+    ex = Settings.exception_notifier
+    if ex
+      tmp = ex.exception_recipient
+      if tmp
+        tmp = Array(tmp.to_s) unless tmp.is_a?(Array)
+      else
+        tmp = []
+      end
+      rs = []
+      tmp.each do |r|
+        r = r.to_s
+        rs.push(r) if (r && r.length > 0)
+      end
+      ex.exception_recipient = rs
+      ex.enabled = rs.size > 0
+      ex.mail_prefix = ex.mail_prefix.to_s if ex.mail_prefix
+      ex.sender_address = ex.sender_address.to_s if ex.sender_address
+    end
+
+    tmp = Settings.secret_token ? Settings.secret_token.to_s : ''
+    # the internet whistles at least 30 characters, we want more
+    if tmp.length >= 64
+      config.secret_token = tmp
+      # Need to keep it consistent - otherwise when the initializer
+      # 'secret_token.rb' gets run, it may override this token and thus all
+      # initializers running after it may use a different token. So ignore any
+      # changes during the initializer phase and restore behavior when finished.
+      def config.secret_token=(value)
```

Comment:
Unused method argument - `value`. If it's necessary, use `_` or `_value` as an argument name to indicate that it won't be used. You can also write as `secret_token=(*)` if you want the method to accept any arguments but don't care about them.